server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1264,7 +1228,7 @@ public void snapshot(final IndexCommit snapshotIndexCommit) {
                         indexIncrementalSize += md.length();
                         // create a new FileInfo
                         BlobStoreIndexShardSnapshot.FileInfo snapshotFileInfo =
-                            new BlobStoreIndexShardSnapshot.FileInfo(fileNameFromGeneration(++generation), md, chunkSize());
+                            new BlobStoreIndexShardSnapshot.FileInfo(DATA_BLOB_PREFIX + UUIDs.randomBase64UUID(), md, chunkSize());
```

Comment:
I believe this violates the snapshot validation where snapshots must be lowercase? (I ran into this with SLM) Is this going to have problems for non-case-sensitive filesystems since different UUIDs could collide there?

Reply:
This blob-name is not related to the name of the snapshot in any way so no such limitations apply here. These are segment data files.

> Is this going to have problems for non-case-sensitive filesystems since different UUIDs could collide there?

I'm not sure that's possible in theory (just from the UUID specification, but I could be wrong here), but I'd say it's unlikely to the point of being impossible even if there were two such UUIDs. But that said, we currently make the same assumption about the safety of these UUIDs for other parts of the blob naming (index and snapshot metadata blobs) as well and so far haven't seen problems arise from that.

Reply:
Okay, thanks for confirming! I only ask because we have to come up with an alternative for SLM (https://github.com/elastic/elasticsearch/blob/1f5811b614765c932732634318d68ce4da4fa6d8/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotLifecyclePolicy.java#L126-L127)